api/services/Ban.js

```diff
@@ -93,13 +93,13 @@ exports.updateBanlist = async function (redToken, username, banlistEntry, friend
     if (knownAlt && lines[i].includes(knownAlt) || lines[i].includes(username)
```

Comment:
This isn't an issue with your proposed solution, of course, but as I'm reviewing this, I'm thinking this line would still cause issues in cases where the username contains underscores since (as far as I'm aware; correct me if I'm wrong, please) the `knownAlt` variable doesn't have the underscores escaped while the `lines[i]` variable still would.

I'm wondering if perhaps it'd be better to just escape the underscores in `knownAlt` and `username` at the start of the function, and then work with those variants. What are your thoughts?

Reply:
I tried to do this escaping but it was somehow breaking `user_regex` and it never matched. I did it opposite way. Assuming that you dont escape underscore while using ban form I just added backslashes at the moment when `_union` is used.